numba/core/ir.py

```diff
@@ -26,7 +26,7 @@ class Loc(object):
     """Source location
 
     """
-    _defmatcher = re.compile(r'def\s+(\w+)\(.*')
+    _defmatcher = re.compile(r'def\s+(\w+)\s*\[.*\]\s*\(.*')
```

Comment:
Although I doubt that it would cause any issues, this regex currently also matches some invalid syntax, e.g.

```
def f[](a: T) -> T:
    assert a != 1
```

Reply:
Your comment made me realize this regex can even be simpler. This regex is expected to run on valid source code. Its purpose is solely for extracting the function name. It doesn't even need to match the closing of the square bracket.